Vungle/src/main/java/com/mopub/mobileads/VungleBanner.java

```diff
@@ -148,30 +157,36 @@ public void run() {
                 public void run() {
                     MoPubLog.log(LOAD_FAILED, ADAPTER_NAME, "Unsupported Banner/Medium rectangle Ad size:  " +
                             "Placement ID:" + mPlacementId);
-                    if (mLoadListener != null) {
-                        mLoadListener.onAdLoadFailed(MoPubErrorCode.ADAPTER_CONFIGURATION_ERROR);
-                    }
+                    mCustomEventBannerListener.onBannerFailed(MoPubErrorCode.ADAPTER_CONFIGURATION_ERROR);
                 }
             });
         }
     }
 
-    private AdConfig.AdSize getVungleAdSize(@NonNull final AdData adData) {
-        final Map<String, String> extras =  adData.getExtras();
+    private AdConfig.AdSize getVungleAdSize(Map<String, Object> localExtras, Map<String, String> serverExtras) {
+        Preconditions.checkNotNull(localExtras);
+        Preconditions.checkNotNull(serverExtras);
 
         AdConfig.AdSize adSizeType = null;
-        int adWidthInDp = adData.getAdWidth() != null ? adData.getAdWidth() : 0;
-        int adHeightInDp = adData.getAdHeight() != null ? adData.getAdHeight() : 0;
+        int adWidthInDp = 0;
+        int adHeightInDp = 0;
 
-        String adUnitFormat = extras.get(ADUNIT_FORMAT);
+        final Object adWidthObject = localExtras.get(AD_WIDTH);
+        if (adWidthObject instanceof Integer) {
+            adWidthInDp = (int) adWidthObject;
+        }
+
+        final Object adHeightObject = localExtras.get(AD_HEIGHT);
+        if (adHeightObject instanceof Integer) {
+            adHeightInDp = (int) adHeightObject;
+        }
+
+        String adUnitFormat = serverExtras.get(ADUNIT_FORMAT);
         if (!TextUtils.isEmpty(adUnitFormat)) {
             adUnitFormat = adUnitFormat.toLowerCase();
         }
-        final boolean isMRECFormat = "medium_rectangle".equals(adUnitFormat);
-        if (isMRECFormat) {
-            if (adWidthInDp >= VUNGLE_MREC.getWidth() && adHeightInDp >= VUNGLE_MREC.getHeight()) {
-                adSizeType = VUNGLE_MREC;
-            }
+        if (adWidthInDp >= VUNGLE_MREC.getWidth() && adHeightInDp >= VUNGLE_MREC.getHeight()) {
```

Comment:
Pls keep `isMRECFormat` logic here, `adWidth` and `adHeight` match VUNGLE_MREC not means this ad unit is MREC ad format.
```
final boolean isMRECFormat = "medium_rectangle".equals(adUnitFormat);
        if (isMRECFormat) {
            if (adWidthInDp >= VUNGLE_MREC.getWidth() && adHeightInDp >= VUNGLE_MREC.getHeight()) {
                adSizeType = VUNGLE_MREC;
            }
        } else {
        ...
        }
```

Reply:
@HuFangshuai The issue is that "medium_rectangle" is missed in old Mopub as far as I remember.

Reply:
Thank @alexander-gubov, yes, MoPub 5.8.0 SDK does not pass `medium_rectangle ` to adapter.